src/viam/ur/module/ur_arm.cpp

```diff
@@ -679,19 +721,43 @@ std::string URArm::status_to_string(UrDriverStatus status) {
 
 // Define the destructor
 URArm::~URArm() {
-    VIAM_SDK_LOG(warn) << "URArm destructor called";
-    current_state_->shutdown.store(true);
-    // stop the robot
-    VIAM_SDK_LOG(info) << "URArm destructor calling stop";
-    stop(ProtoStruct{});
-    // disconnect from the dashboard
-    if (current_state_->dashboard) {
-        VIAM_SDK_LOG(info) << "URArm destructor calling dashboard->disconnect()";
-        current_state_->dashboard->disconnect();
+    VIAM_SDK_LOG(warn) << "URArm destructor called, shutting down";
+    shutdown_();
+    VIAM_SDK_LOG(warn) << "URArm destroyed";
+}
+
+void URArm::shutdown_() noexcept {
+    try {
+        VIAM_SDK_LOG(warn) << "URArm shutdown called";
+        if (current_state_) {
+            const auto destroy_state = make_scope_guard([&] { current_state_.reset(); });
```

Comment:
This ensures that we go back to a state where we can call `startup_` (current_state_ is empty) as a result of calling `shutdown_`.